yb-voyager/src/srcdb/main_test.go

```diff
@@ -0,0 +1,165 @@
+)
+
+type TestDB struct {
+	testcontainers.TestContainer
+	*Source
+}
+
+var (
+	testPostgresSource   *TestDB
+	testOracleSource     *TestDB
+	testMySQLSource      *TestDB
+	testYugabyteDBSource *TestDB
+)
+
+func TestMain(m *testing.M) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// setting source db type, version and defaults
+	postgresContainer := testcontainers.NewTestContainer("postgresql", nil)
+	// TODO: handle error
+	err := postgresContainer.Start(ctx)
+	if err != nil {
+		utils.ErrExit("Failed to start postgres container: %v", err)
+	}
+	host, port, err := postgresContainer.GetHostPort()
+	if err != nil {
+		utils.ErrExit("%v", err)
+	}
+	testPostgresSource = &TestDB{
+		TestContainer: postgresContainer,
+		Source: &Source{
+			DBType:    "postgresql",
+			DBVersion: postgresContainer.GetConfig().DBVersion,
+			User:      postgresContainer.GetConfig().User,
+			Password:  postgresContainer.GetConfig().Password,
+			Schema:    postgresContainer.GetConfig().Schema,
+			DBName: postgresContainer.GetConfig().DBName,
+			Host:      host,
+			Port:      port,
+			SSLMode:   "disable",
+		},
+	}
+	err = testPostgresSource.DB().Connect()
+	if err != nil {
+		utils.ErrExit("Failed to connect to postgres database: %w", err)
+	}
+	defer testPostgresSource.DB().Disconnect()
+
+	oracleContainer := testcontainers.NewTestContainer("oracle", nil)
+	_ = oracleContainer.Start(ctx)
+	host, port, err = oracleContainer.GetHostPort()
+	if err != nil {
+		utils.ErrExit("%v", err)
+	}
+	testOracleSource = &TestDB{
+		TestContainer: oracleContainer,
+		Source: &Source{
+			DBType:    "oracle",
+			DBVersion: oracleContainer.GetConfig().DBVersion,
+			User:      oracleContainer.GetConfig().User,
+			Password:  oracleContainer.GetConfig().Password,
+			Schema:    oracleContainer.GetConfig().Schema,
+			DBName:    oracleContainer.GetConfig().DBName,
+			Host:      host,
+			Port:      port,
+		},
+	}
+
+	err = testOracleSource.DB().Connect()
+	if err != nil {
+		utils.ErrExit("Failed to connect to oracle database: %w", err)
+	}
+	defer testOracleSource.DB().Disconnect()
+
+	mysqlContainer := testcontainers.NewTestContainer("mysql", nil)
+	_ = mysqlContainer.Start(ctx)
+	host, port, err = mysqlContainer.GetHostPort()
+	if err != nil {
+		utils.ErrExit("%v", err)
+	}
+	testMySQLSource = &TestDB{
+		TestContainer: mysqlContainer,
+		Source: &Source{
+			DBType:    "mysql",
+			DBVersion: mysqlContainer.GetConfig().DBVersion,
+			User:      mysqlContainer.GetConfig().User,
+			Password:  mysqlContainer.GetConfig().Password,
+			Schema:    mysqlContainer.GetConfig().Schema,
+			DBName:    mysqlContainer.GetConfig().DBName,
+			Host:      host,
+			Port:      port,
+			SSLMode:   "disable",
+		},
+	}
+
+	err = testMySQLSource.DB().Connect()
+	if err != nil {
+		utils.ErrExit("Failed to connect to mysql database: %w", err)
+	}
+	defer testMySQLSource.DB().Disconnect()
```

Comment:
nit: since you're only connecting to test that the connections are working, you don't really need the "defer", you can disconnect at this point itself.

Reply:
Hmm, good point. How is it working now then for each test that uses the test containers?

Reply:
> How is it working now then for each test that uses the test containers?

didn't get you exactly, i think all the tests(`srcdb` or `tgtdb` packages) i have added is following this. 
Are there any existing tests which don't follow this but still works?

Reply:
My understanding is that `TestMain` runs before any other test in the package. 
And if `TestMain` has a defer to disconnect the DB, then how is it that in `TestPostgresGetAllTableNames`, you are able to use that container to execute SQLs? It should have been disconnected by then, right?

Reply:
@makalaaneesh in `TestMain` we have to call `m.Run()` which executes all the tests of that package.
So here defer will be invoked after all the tests for that package has run.
like you can i added `terminateAllContainers` after that.`

I will add one comment in the code for this.